pkg/apis/delivery/v1alpha1/policy_state_types.go

```diff
@@ -45,17 +48,40 @@ var (
 	_ duckv1.KRShaped = (*PolicyState)(nil)
 )
 
-// PolicyStateSpec holds the desired state of the PolicyState
-// Should be set by reconciler, and used by webhook to write Route appropriately
+const (
+	// PolicyStateConditionRouteConfigured is set to false if any failure prevents
+	// PolicyState.Spec from being written to Route.Spec
+	PolicyStateConditionRouteConfigured apis.ConditionType = "RouteConfigured"
+)
+
+// TrafficTarget reuses Route TrafficTarget
+type TrafficTarget v1.TrafficTarget
+
+// PolicyStateSpec holds the desired routing spec computed by reconciler
+// Should be set by reconciler, and set by webhook to write Route appropriately
 type PolicyStateSpec struct {
-	// TODO: implement policy state spec
+	// Traffic specifies how to distribute traffic over a collection of
+	// revisions and configurations.
+	Traffic []TrafficTarget `json:"traffic,omitempty"`
 }
 
 // PolicyStateStatusFields holds the fields of PolicyState's status that
 // are not generally shared.  This is defined separately and inlined so that
 // other types can readily consume these fields via duck typing.
 type PolicyStateStatusFields struct {
-	// TODO: implement policy state status
+	// NextUpdateTimestamp specifies the next time when this PolicyState spec should be updated
+	// it is used in conjunction with EnqueueAfter to help reconciler enforce time-based policies
+	// it also helps prevent unexpected rollout behavior when controller restarts, etc.
+	// optional because when a rollout is completed there is no more future updates to be done
+	NextUpdateTimestamp *metav1.Time `json:"nextUpdateTimestamp,omitempty"`
+
+	// StaticRevisions specifies which Revisions currently have static percentages
```

Comment:
I think this should be removed until there are active plans to support static revisions.

Reply:
I commented this out

Reply:
This should be removed entirely. It could be a while until this is addressed. We can add it back then.

Reply:
deleted